packages/commons/test/FileAxeScriptProviderTest.cs

```diff
@@ -0,0 +1,69 @@
+using NUnit.Framework;
+using Deque.AxeCore.Commons;
+using Moq;
+using FluentAssertions;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Deque.AxeCore.Commons.Test
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class FileAxeScriptProviderTest
+    {
+        private static readonly string testContext = TestContext.CurrentContext.TestDirectory.ToString();
+        private readonly string basePath = testContext.Split(new string[] { "test" }, StringSplitOptions.None)[0];
+        
+        [Test]
+        public void ConstructorPassedValidFile()
+        {
+            var absolutePath = Path.Combine(basePath, "src", "Resources", "sampleFile.txt");
+            
+            var scriptProvider = new FileAxeScriptProvider(absolutePath);
+            scriptProvider.Should().NotBeNull();
+            var readResult = scriptProvider.GetScript();
+            readResult.Should().Be("sample output");
+        }
+
+        [Test]
+        public void ConstructorPassedNullOrEmptyString()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var scriptProvider = new FileAxeScriptProvider(null);                
+                scriptProvider.Should().NotBeNull();
+            });
+            
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var scriptProvider = new FileAxeScriptProvider("");                
```

Comment:
I think this case suggests that the product code is wrong; passing a non-null but invalid input should result in an `ArgumentException`, not an `ArgumentNullException`